Index: App_Python_Wrapper.cpp
===================================================================
--- App_Python_Wrapper.cpp	(revision 15341)
+++ App_Python_Wrapper.cpp	(revision 15358)
@@ -346,6 +346,7 @@
 
 
 #include <string.h>
+#include <map>
 
 #ifdef __cplusplus
 extern "C" {
@@ -555,16 +556,33 @@
   swig_cast_info *cast = ti->cast;
   /* if (ti->clientdata == clientdata) return; */
   ti->clientdata = clientdata;
-  
+
+  // FIX: Check if endless loop exists in the list and mark its position
+  std::map<swig_cast_info*, int> castMap;
+
+
   while (cast) {
+    // FIX: check if this address is already in the map
+    std::map<swig_cast_info *, int>::iterator it = castMap.find(cast);
+    if (it != castMap.end())
+    {
+      break;
+    }
+    else
+    {
+      castMap.insert(it, std::map<swig_cast_info *, int>::value_type(cast, -1));
+    }
+
     if (!cast->converter) {
       swig_type_info *tc = cast->type;
       if (!tc->clientdata) {
-	SWIG_TypeClientData(tc, clientdata);
+        SWIG_TypeClientData(tc, clientdata);
+
       }
     }    
     cast = cast->next;
   }
+
 }
 SWIGRUNTIME void
 SWIG_TypeNewClientData(swig_type_info *ti, void *clientdata) {
