diff --git a/xbmc/lib/libbluray/src/libbluray/bluray.c b/xbmc/lib/libbluray/src/libbluray/bluray.c
index 1e6133e..5e6fd98 100644
--- a/xbmc/lib/libbluray/src/libbluray/bluray.c
+++ b/xbmc/lib/libbluray/src/libbluray/bluray.c
@@ -390,7 +390,7 @@ static int _read_block(BLURAY *bd, BD_STREAM *st, uint8_t *buf)
 #endif
 
                 }
-
+#if 0
                 /* Check TP_extra_header Copy_permission_indicator. If != 0, unit is still encrypted. */
                 if (buf[0] & 0xc0) {
                     BD_DEBUG(DBG_BLURAY | DBG_CRIT,
@@ -398,7 +398,7 @@ static int _read_block(BLURAY *bd, BD_STREAM *st, uint8_t *buf)
                     _queue_event(bd, (BD_EVENT){BD_EVENT_ENCRYPTED, 0});
                     return 0;
                 }
-
+#endif
                 BD_DEBUG(DBG_STREAM, "Read unit OK! (%p)\n", bd);
 
                 return 1;
